qa/rpc-tests/wallet_orchard.py

```diff
@@ -121,25 +124,36 @@ def run_test(self):
                 {'pools': {'orchard': {'valueZat': Decimal('100000000')}}, 'minimum_confirmations': 1}, 
                 self.nodes[3].z_getbalanceforaccount(acct3))
 
-        # TODO: enable after wallet persistence
-        # # Rejoin the network. 
-        # self.join_network()
+        # Check that the mempools are empty
+        for i in range(self.num_nodes):
+            assert_equal(set([]), set(self.nodes[i].getrawmempool()))
+
+        # Reconnect the nodes; nodes 2 and 3 will re-org to node 0's chain.
+        print("Re-joining the network so that nodes 2 and 3 reorg")
+        # We can't use `self.join_network()` because the the nodes's mempools fail to synchronize on restart
+        assert self.is_network_split
+        stop_nodes(self.nodes)
+        wait_bitcoinds()
+        self.nodes = self.setup_nodes()
+        connect_nodes_bi(self.nodes, 1, 2)
+        sync_blocks(self.nodes[1:3])
+        connect_nodes_bi(self.nodes, 0, 1)
+        connect_nodes_bi(self.nodes, 2, 3)
+        self.is_network_split = False
+        sync_blocks(self.nodes)
 
         # split 0/1's chain should have won, so their wallet balance should be consistent
         assert_equal(
                 {'pools': {'orchard': {'valueZat': Decimal('2000000000')}}, 'minimum_confirmations': 1}, 
                 self.nodes[1].z_getbalanceforaccount(acct1))
 
-        # TODO: enable after wallet persistence
-        # # split 2/3's chain should have been rolled back, so their txn should have been
-        # # un-mined
-        # assert_equal(
-        #         {'pools': {'sapling': {'valueZat': Decimal('1000000000')}}, 'minimum_confirmations': 1}, 
-        #         self.nodes[2].z_getbalanceforaccount(acct2))
```

Comment:
It would be useful (either in this or a subsequent PR) to check that:
- `acct2`'s balance is 0 with the default minconf (the transaction is still in the mempool and thus spent from node 2's perspective), and 9 ZEC with minconf of 0.
- `acct3`'s balance is non-zero with minconf 0 (we already have the default minconf check below).